frontend/src/components/MarketCardList.js

```diff
@@ -3,41 +3,45 @@ import React, { useState, useLayoutEffect } from 'react';
 import '../styles/MarketCardList.css';
 import { fetchUpcomingMarkets } from './Service';
 import { Link } from 'react-router-dom';
+import ErrorBody from './ErrorBody';
 
 const MarketCardList = () => {
   const [upcomingMarkets, setUpcomingMarkets] = useState([]);
+  const [error, setError] = useState(null);
 
   const getUpcomingMarkets = async () => {
     const result = await fetchUpcomingMarkets();
     setUpcomingMarkets(result);
   };
 
   useLayoutEffect(() => {
-    getUpcomingMarkets();
+    getUpcomingMarkets().catch((err) => setError(err.message));
   }, []);
 
   return (
     <div className='card-list'>
-      {upcomingMarkets.map((market) => {
-        return (
-          <div key={market.id}>
-            <Link
-              to={`/piacok/${market.id}`}
-              style={{ textDecoration: 'none' }}
-            >
-              <MarketCard
+      <ErrorBody error={error} />
```

Comment:
ErrorBody-t csak akkor jelenitsuk meg ha van error. igy, hogy itt nincs feltetel, minden esetben ki fog ertekelodni a ErrorBody functional component ami teljesen felesleges, valamint egy ures div is bekerul a DOM-ba szinten feleslegesen. `<div className='error-body'>`